### PR TITLE
doc: syscall: add K-V syscall doc

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This pull request adds syscall documentation for the KV interface.

[Rendered](https://github.com/tock/tock/blob/syscall-kv-doc/doc/syscalls/50003_key_value.md)

### Testing Strategy

na

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: doc/syscalls/50003_key_value.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  - `RESERVE`: Error in the driver, requesting process not set or value allow
+    buffer not set.
+  - `SIZE`: Key too long or value too long.
+  - `INVAL`: Incorrect permissions for the app.
+
+- ### Command number: `3`
+
+  **DELETE**. Delete a key-value pair from the database.
+
+  Use RO allow 0 to set the key and RO allow 1 to set the value.
```
Comment (alistair23) [inline doc/syscalls/50003_key_value.md:74]: We don't need RO allow 1

File: doc/syscalls/50003_key_value.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+  #### Returns
+
+  `SUCCESS` if it exists, otherwise `NODEVICE`.
+
+- ### Command number: `1`
+
+  **GET**. Retrieve a value from the store based on the provided key.
+
+  Use RO allow 0 to set the key.
```
Comment (alistair23) [inline doc/syscalls/50003_key_value.md:31]: hashed key?

**Review by alistair23 — CHANGES_REQUESTED** [review]

File: doc/syscalls/50003_key_value.md (hunk 2 above)
Comment (bradjc) [reply doc/syscalls/50003_key_value.md:31]: No it's just any key.

File: doc/syscalls/50003_key_value.md (hunk 1 above)
Comment (bradjc) [reply doc/syscalls/50003_key_value.md:74]: The interface currently uses it.

Comment (bradjc) [reply doc/syscalls/50003_key_value.md:74]: Oh my bad.

Comment (bradjc) [reply doc/syscalls/50003_key_value.md]: Fixed.

**Review by alistair23 — APPROVED** [review]

**Review by lschuermann — APPROVED** [review]

**Comment by bradjc** [comment]
This seems like an easy merge?

**Pull request merged**